src/main/java/echo/parser/Parser.java

```diff
@@ -0,0 +1,280 @@
+import echo.command.FindCommand;
+import echo.command.HelpCommand;
+import echo.command.ListCommand;
+import echo.command.MarkCommand;
+import echo.command.TodoCommand;
+import echo.command.UnmarkCommand;
+import echo.utils.EchoException;
+
+/**
+ * Parser encapsulates a parser to deal with making sense of the user command.
+ */
+public class Parser {
+
+    /** Is user exited */
+    private static boolean isExit = false;
+
+    /**
+     * Parses user command and determines action to perform.
+     *
+     * @param input Input from user.
+     * @return Command to be executed.
+     *
+     * @throws EchoException If input is invalid.
+     */
+    public static Command parse(String input) throws EchoException {
+        try {
+            String[] splitSpace = input.split(" ");
+            String command = splitSpace[0];
+
+            switch (command) {
+            case ListCommand.COMMAND:
+                return prepareListCommand();
+            case TodoCommand.COMMAND:
+                // Fallthrough
+            case DeadlineCommand.COMMAND:
+                // Fallthrough
+            case EventCommand.COMMAND:
+                String[] splitSlash = input.split("/");
+
+                String desc = getDescription(splitSlash[0], command);
+
+                // If second input (description) is not specified.
+                if (splitSpace.length == 1) {
+                    throw new EchoException(String.format("The description of a %s cannot be empty.", input));
+                }
+
+                if (command.equals(TodoCommand.COMMAND)) {
+                    return prepareTodoCommand(desc);
+                }
+
+                if (!input.contains("/")) {
+                    throw new EchoException("Please specify the time of the task: \n"
+                            + "<command> <description> /by <yyyy-mm-dd> <24hr time>");
+                }
+
+                try {
+                    if (command.equals(DeadlineCommand.COMMAND)) {
+                        return prepareDeadlineCommand(input, desc, getLocalDateTime(splitSlash));
+                    } else {
+                        return prepareEventCommand(input, desc, getLocalDateTime(splitSlash));
+                    }
+                } catch (DateTimeParseException e) {
+                    throw new EchoException("Invalid format. Follow the <yyyy-mm-dd> <24hr time> format."
+                            + "\n" + "E.g. 2019-10-15 1800");
+                }
+            case MarkCommand.COMMAND:
+                // Fallthrough
+            case UnmarkCommand.COMMAND:
+                // Fallthrough
+            case DeleteCommand.COMMAND:
+                // If second input (task number) is not specified.
+                if (splitSpace.length == 1) {
+                    throw new EchoException("Please specify the task number. Eg. mark 1");
+                }
+
+                // Parse second input into an int, throws NumberFormatException if it's not an int.
```

Comment:
The in-text comments have been really helpful in understanding your code. Just a small note to take, "Parses" might be more appropriate! :)